src/wp-includes/block-template-utils.php

```diff
@@ -317,6 +338,17 @@ function _get_block_templates_files( $template_type ) {
 				// Subtract ending '.html'.
 				-5
 			);
+
+			// Skip this item if its slug doesn't match any of the slugs to include.
+			if ( ! empty( $slugs_to_include ) && ( false === array_search( $template_slug, $slugs_to_include, true ) ) ) {
+				continue;
+			}
+
+			// Skip this item if its slug matches any of the slugs to skip.
+			if ( ! empty( $slugs_to_skip ) && ( false !== array_search( $template_slug, $slugs_to_skip, true ) ) ) {
```

Comment:
```suggestion
			if ( ! empty( $slugs_to_skip ) && isset( $slugs_to_skip[ $template_slug ] ) ) {
```

Check similar comment above.

Reply:
Check https://github.com/WordPress/wordpress-develop/pull/4097#discussion_r1150462689

Reply:
https://github.com/WordPress/wordpress-develop/pull/4097/commits/d79f949dc839e87b02911298358346d048201d75